app/src/test/java/tech/ula/utils/ValidationUtilityTest.kt

```diff
@@ -0,0 +1,229 @@
+                "tty",
+                "users",
+                "utmp",
+                "uucp",
+                "uuidd",
+                "vboxusers",
+                "video",
+                "voice",
+                "wheel",
+                "www-data"
+        )
+
+        whenever(resources.getStringArray(R.array.blacklisted_usernames))
+                .thenReturn(arrayOf("adm", "audio"))
+        whenever(resources.getString(R.string.error_empty_field))
+                .thenReturn("Each field must be entered!")
+        whenever(resources.getString(R.string.error_username_invalid_characters))
+                .thenReturn("Characters in username must be 1 to 30 characters with lowercase letters and/or numbers.")
+        whenever(resources.getStringArray(R.array.blacklisted_usernames))
+                .thenReturn(blacklistedUsernames)
+        whenever(resources.getString(R.string.error_password_invalid))
+                .thenReturn("Password has invalid characters")
+        whenever(resources.getString(R.string.error_vnc_password_invalid))
+                .thenReturn("VNC Password has invalid characters")
+        whenever(resources.getString(R.string.error_vnc_password_length_incorrect))
+                .thenReturn("VNC Password must be between 6 to 8 characters!")
+    }
+
+    @Test
+    fun `Username is empty`() {
+        val username = ""
+
+        credential = validationUtility.validateUsername(username)
+        assertFalse(credential.credentialIsValid)
+        assertEquals(credential.errorMessage, "Each field must be entered!")
+    }
+
+    @Test
+    fun `Username is capitalized`() {
+        val username = "A"
+        credential = validationUtility.validateUsername(username)
+        assertFalse(credential.credentialIsValid)
+        assertEquals(credential.errorMessage, "Characters in username must be 1 to 30 characters with lowercase letters and/or numbers.")
+    }
+
+    @Test
+    fun `Username has capital letters`() {
+        val username = "abC"
+        credential = validationUtility.validateUsername(username)
+        assertFalse(credential.credentialIsValid)
+        assertEquals(credential.errorMessage, "Characters in username must be 1 to 30 characters with lowercase letters and/or numbers.")
+    }
+
+    @Test
+    fun `Username starts with numbers`() {
+        val username = "123abc"
+        credential = validationUtility.validateUsername(username)
+        assertFalse(credential.credentialIsValid)
+        assertEquals(credential.errorMessage, "Characters in username must be 1 to 30 characters with lowercase letters and/or numbers.")
+    }
+
+    @Test
+    fun `Username starts with underscore`() {
+        val username = "_123abc"
+        credential = validationUtility.validateUsername(username)
+        assertTrue(credential.credentialIsValid)
+        assertEquals(credential.errorMessage, "")
+    }
+
+    @Test
+    fun `Username is in blacklist`() {
+        val username = "root"
```

Comment:
So for example the array the stub above returns could literally just be `arrayOf("root")`